### ISSUE TITLE
Position Component with Box Decoration

### ISSUE THREAD

**Issue by stuartrapop** [flame-engine/flame#3374]
### Problem to solve

The ability to style a rectangular position component with radius for border, border radius, border color, background color, border shadow (blur, color and offset), etc. and maybe hover over color. 

I think this would be very useful for a lot of people, although this feature is definitely not urgent as the desired result can already be achieved (albeit with a bit of redundant code). 

### Proposal

Maybe a mixin on PositionComponent could make sense ? 

### More information

https://discord.com/channels/509714518008528896/1309774265930879006/1309774265930879006

### Other

- [X] Are you interested in working on a PR for this?

**Comment by spydon** [comment]
Since `PositionComponent`s most of the time aren't rectangular in how they are drawn on the canvas it might be a bit confusing to have this feature connected to the `PositionComponent`, a better candidate would be the `RectangleComponent`.

**Comment by stuartrapop** [comment]
> Since `PositionComponent`s most of the time aren't rectangular in how they are drawn on the canvas it might be a bit confusing to have this feature connected to the `PositionComponent`, a better candidate would be the `RectangleComponent`.

Makes sense !

**Comment by spydon** [comment]
Do you want to work on this @stuartrapop? :)

**Comment by stuartrapop** [comment]
> Do you want to work on this @stuartrapop? :)

Yes, may I message you in mp for some guidance? I am eager to do this but started and quickly got stuck.
